backend/src/main/java/com/redhat/cloud/notifications/db/repositories/BehaviorGroupRepository.java

```diff
@@ -395,6 +395,76 @@ public void updateEventTypeBehaviors(String orgId, UUID eventTypeId, Set<UUID> b
         }
     }
 
+    /**
+     * Deletes the specified behavior group from the event type, by deleting the relation in the "event_type_behavior"
+     * table.
+     * @param eventTypeUuid the event type to remove the behavior group from.
+     * @param behaviorGroupUuid the behavior group to remove.
+     * @param orgId the org id to make sure that the caller has visibility on that behavior group, so that they don't
+     *              delete unowned relations.
+     */
+    @Transactional
+    public void deleteBehaviorGroupFromEventType(final UUID eventTypeUuid, final UUID behaviorGroupUuid, final String orgId) {
+        // First check that the specified event type exists.
+        final var eventTypeExistsQuery =
+                "SELECT " +
+                    "1 " +
+                "FROM " +
+                    "EventType AS e " +
+                "WHERE " +
+                    "e.id = :eventTypeUuid";
+
+        try {
+            this.entityManager.createQuery(eventTypeExistsQuery)
+                .setParameter("eventTypeUuid", eventTypeUuid)
+                .getSingleResult();
+        } catch (final NoResultException e) {
+            throw new BadRequestException("the specified event type does not exist");
+        }
+
+        // Second, check that the behavior group also exists.
+        final var behaviorGroupExistsQuery =
+                "SELECT " +
+                        "1 " +
+                "FROM " +
+                    "BehaviorGroup AS bg " +
+                "WHERE " +
+                    "bg.id = :behaviorGroupUuid " +
+                "AND " +
+                    "bg.orgId = :orgId";
+
+        try {
+            this.entityManager.createQuery(behaviorGroupExistsQuery)
+                .setParameter("behaviorGroupUuid", behaviorGroupUuid)
+                .setParameter("orgId", orgId)
+                .getSingleResult();
+        } catch (final NoResultException e) {
+            throw new BadRequestException("the specified behavior group does not exist");
+        }
+
+        // Finally attempt to delete the relation.
+        final var deleteFromEventTypeQuery =
+            "DELETE FROM " +
+                "event_type_behavior AS etb " +
+            "USING " +
+                "behavior_group AS bg " +
+            "WHERE " +
+                "etb.behavior_group_id = bg.id " +
+            "AND " +
+                "etb.behavior_group_id = :behaviorGroupUuid " +
+            "AND " +
+                "bg.org_id = :orgId";
+
+        javax.persistence.Query query = this.entityManager.createNativeQuery(deleteFromEventTypeQuery);
```

Comment:
Does that query have to be native?

Reply:
The thing is that in order to join the `BehaviorGroup` table and make sure that we are deleting the relation from the right tenant, the `USING` keyword needs to be used, and with a `createQuery` Hibernate complaints...